### ISSUE TITLE
legend() ignores box.col argument

### ISSUE THREAD

**Issue by ctdenney** [rstudio/rstudio#11525]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.02.3 Build 492
    OS Version      :  Windows 11
    R Version       :  4.2.1

### Steps to reproduce the problem
1. create a plot using base R graphics
2. Add a legend to this plot, with bty = 'o' and box.col set to a color other than 'black'
Reproducible example:
```
plot(0~1)
legend('bottomleft', legend = 'test', pch = 19, col = 'black', box.col = 'green', inset = 0.1)
```

### Describe the problem in detail
the legend function ignores the box.col argument. Problem exists in the plots frame, or when exported using png(), jpeg(), or pdf(). Problem does not exist if code is run in the regular R GUI (include exports).
### Describe the behavior you expected
I expect that, when bty = 'o', that box.col will change the color of the border around the legend.

- [X ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [X ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@ctdenney Thank you for raising this! I'm unable to reproduce this though
- RStudio Desktop 2022.02.3+492 and 2022.07.0-preview+453 on MacOS 12.5
- RStudio Desktop 2022.02.3+492 and 2022.07.0-preview+453 on Windows 11 
- RStudio Server 2022.07.0-preview+453 on Ubuntu 20.04 via Chrome on MacOS 12.5

I'll mark it for review, though we'll probably need more information (TBD).

**Comment by ctdenney** [comment]
@ronblum Your comment got me to go and revisit, and I discovered that this is not an RStudio bug. While I did go to the effort of trying in the R GUI, I did not restart my R session inside RStudio, which I did today, and led me to discover that the problem is one with a package hijacking the legend function in a way that it will still _mostly_ operate as expected, and, despite the hijacked version also having a box.col argument, it's non functional. The base version of legend() is fine.
